packages/rest/__tests__/RequestHandler.test.ts

```diff
@@ -330,33 +367,132 @@ test('Handle unexpected 429', async () => {
 });
 
 test('Handle unexpected 429 cloudflare', async () => {
+	mockPool
+		.intercept({
+			path: genPath('/unexpected-cf'),
+			method: 'GET',
+		})
+		.reply(() => {
+			if (unexpected429cf) {
+				unexpected429cf = false;
+
+				return {
+					statusCode: 429,
+					data: '',
+					responseOptions: {
+						headers: {
+							'retry-after': '1',
+						},
+					},
+				};
+			}
+
+			return {
+				statusCode: 200,
+				data: { test: true },
+				responseOptions,
+			};
+		})
+		.times(2); // twice because it re-runs the request after first 429
+
 	const previous = Date.now();
 	expect(await api.get('/unexpected-cf')).toStrictEqual({ test: true });
 	expect(Date.now()).toBeGreaterThanOrEqual(previous + 1000);
 });
 
+test('Handle global rate limits', async () => {
+	mockPool
+		.intercept({
+			path: genPath('/triggerGlobal'),
+			method: 'GET',
+		})
+		.reply(() => ({
+			data: { global: true },
+			statusCode: 200,
+			responseOptions,
+		}));
+
+	mockPool
+		.intercept({
+			path: genPath('/regularRequest'),
+			method: 'GET',
+		})
+		.reply(() => ({
+			data: { test: true },
+			statusCode: 200,
+			responseOptions,
+		}));
+
+	const earlier = performance.now();
+	expect(await api.get('/triggerGlobal')).toStrictEqual({ global: true });
+	expect(await api.get('/regularRequest')).toStrictEqual({ test: true });
+	expect(performance.now()).toBeGreaterThanOrEqual(earlier + 1);
+});
+
 test('Handle temp server outage', async () => {
+	mockPool
+		.intercept({
+			path: genPath('/temp'),
+			method: 'GET',
+		})
+		.reply(() => {
+			if (serverOutage) {
+				serverOutage = false;
+
+				return {
+					statusCode: 500,
+					data: '',
+				};
+			}
+
+			return {
+				statusCode: 200,
+				data: { test: true },
+				responseOptions,
+			};
+		})
+		.times(2);
+
 	expect(await api.get('/temp')).toStrictEqual({ test: true });
 });
 
 test('perm server outage', async () => {
+	mockPool
+		.intercept({
+			path: genPath('/outage'),
+			method: 'GET',
+		})
+		.reply(500, '', responseOptions)
+		.times(4);
+
 	const promise = api.get('/outage');
 	await expect(promise).rejects.toThrowError();
 	await expect(promise).rejects.toBeInstanceOf(HTTPError);
 });
 
 test('server responding too slow', async () => {
-	const promise = api.get('/slow');
-	await expect(promise).rejects.toThrowError('The user aborted a request.');
-}, 10000);
+	mockPool
+		.intercept({
+			path: genPath('/slow'),
+			method: 'GET',
+		})
+		.reply(200, '')
+		.delay(8000)
```

Comment:
This is....*very* slow, waiting for the old test already felt like ages, lets not wait longer (remember we try twice). If needed you can add another rest handler that has a much shorter timeout

Reply:
That is the only timeout that works 100%, otherwise the test was flaky maybe 30-40% of the time.

Reply:
what i mean is add another api handler with timeout: 250 set in the options so that you can decrease this

Reply:
that makes sense, will do

Reply:
done in fa21eb416ac1f80f349e1be0be3d178b9fb1b3f2